apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,61 +1,36 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgStyle, NgTemplateOutlet } from '@angular/common';
+import { Component, ContentChild, Input, TemplateRef } from '@angular/core';
 import { ListItemComponent } from '../list-item/list-item.component';
-
 @Component({
   selector: 'app-card',
   template: `
     <div
       class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
-      [class]="customClass">
-      <img
-        *ngIf="type === CardType.TEACHER"
-        src="assets/img/teacher.png"
-        width="200px" />
-      <img
-        *ngIf="type === CardType.STUDENT"
-        src="assets/img/student.webp"
-        width="200px" />
-
+      [ngStyle]="{ 'background-color': customClass }">
+      <ng-content select="img"></ng-content>
       <section>
-        <app-list-item
-          *ngFor="let item of list"
-          [name]="item.firstName"
-          [id]="item.id"
-          [type]="type"></app-list-item>
+        @for (item of list; track item.id) {
+          <app-list-item [id]="item.id">
+            <ng-container
+              *ngTemplateOutlet="
+                deleteButton;
+                context: {
+                  $implicit: item.id,
+                  name: item.name,
+                  firstName: item.firstName
+                }
+              "></ng-container>
+          </app-list-item>
+        }
       </section>
-
-      <button
-        class="rounded-sm border border-blue-500 bg-blue-300 p-2"
-        (click)="addNewItem()">
-        Add
-      </button>
+      <ng-content select="button"></ng-content>
     </div>
   `,
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [ListItemComponent, NgTemplateOutlet, NgStyle],
 })
 export class CardComponent {
   @Input() list: any[] | null = null;
-  @Input() type!: CardType;
   @Input() customClass = '';
-
-  CardType = CardType;
-
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore,
-  ) {}
-
-  addNewItem() {
-    if (this.type === CardType.TEACHER) {
-      this.teacherStore.addOne(randTeacher());
-    } else if (this.type === CardType.STUDENT) {
-      this.studentStore.addOne(randStudent());
-    }
-  }
+  @ContentChild('deleteButton') deleteButton!: TemplateRef<any>;
```

Comment:
You can check out [my PR](https://github.com/tomalaforge/angular-challenges/pull/576/files/361d3ee8f51f2f8ce6303b98ad665717bc23d038#diff-af224cd0c2986716242c984f2b73acc35291c167462cb08223ec20f1b46eac47R45) to see how to get rid of the `any` type.

Reply:
Thanks for your comment and sharing your code, I didn't think to apply a directive but have read the reasoning behind it and implemented it in my latest commit.